File: fiber-product/utils/response.go
package utils

import "github.com/gofiber/fiber/v3"

type Meta struct {
	Status  string `json:"status"`
	Message string `json:"message"`
}

type ApiResponse struct {
	Meta    Meta        `json:"meta"`
	Results interface{} `json:"results"`
}

// SendSuccessAll for arrays/slices
func SendSuccessAll(c fiber.Ctx, data interface{}) error {
	return c.Status(fiber.StatusOK).JSON(ApiResponse{
		Meta: Meta{
			Status:  "Success",
			Message: "Success Retrieve All Data",
		},
		Results: data,
	})
}

// SendSuccessOne for single objects
func SendSuccessOne(c fiber.Ctx, data interface{}) error {
	return c.Status(fiber.StatusOK).JSON(ApiResponse{
		Meta: Meta{
			Status:  "Success",
			Message: "Success Retrieve One Data",
		},
		Results: data,
	})
}

// SendSuccessCustom for specific messages
func SendSuccessCustom(c fiber.Ctx, data interface{}, message string) error {
	return c.Status(fiber.StatusOK).JSON(ApiResponse{
		Meta: Meta{
			Status:  "Success",
			Message: message,
		},
		Results: data,
	})
}

// SendErrorNotFound
func SendErrorNotFound(c fiber.Ctx, message string) error {
	if message == "" {
		message = "Data Not Found"
	}
	return c.Status(fiber.StatusNotFound).JSON(ApiResponse{
		Meta: Meta{
			Status:  "Not Found",
			Message: message,
		},
		Results: nil,
	})
}

// SendError generic error
func SendError(c fiber.Ctx, statusCode int, statusStr string, message string) error {
	return c.Status(statusCode).JSON(ApiResponse{
		Meta: Meta{
			Status:  statusStr,
			Message: message,
		},
		Results: nil,
	})
}

File: rust-message/src/utils/response.rs
use axum::{response::IntoResponse, Json};
use serde::Serialize;

#[derive(Serialize)]
pub struct ApiResponse<T: Serialize> {
    pub meta: Meta,
    pub results: Option<T>,
}

#[derive(Serialize)]
pub struct Meta {
    pub status: String,
    pub message: String,
}

pub fn success_response<T: Serialize>(data: T, message: &str) -> impl IntoResponse {
    Json(ApiResponse {
        meta: Meta {
            status: "Success".to_string(),
            message: message.to_string(),
        },
        results: Some(data),
    })
}

pub fn success_response_all<T: Serialize>(data: T) -> impl IntoResponse {
    success_response(data, "Success Retrieve All Data")
}

pub fn success_response_one<T: Serialize>(data: T) -> impl IntoResponse {
    success_response(data, "Success Retrieve One Data")
}
